### ISSUE TITLE
export to interface

### ISSUE THREAD

**Issue by tatethurston** [tatethurston/nextjs-routes#38]
This so cool. I will use it, but can you export it to an interface too? Sometimes need use like this

```
const myText=`this is a link ${ROUTES.ITEM({itemId:"abcdefg"})} `

```

_Originally posted by @IRediTOTO in https://github.com/vercel/next.js/discussions/28622#discussioncomment-3126130_

**Comment by tatethurston** [comment]
@IRediTOTO Thanks for the kind words. Could you expand on what ROUTES.ITEMS. returns? Is it the pathname?

And do you want an interface to type the arguments to ROUTES.ITEMS or to have ROUTES.ITEMS generated as well?

**Comment by IRediTOTO** [comment]
Hi, I mean
```
const ROUTES={
ITEM:(itemId)=>{
return `/just-a-link/${itemId}`
}}

// same as
/just-a-link/[itemId]



```

**Comment by tatethurston** [comment]
I think there is some overlap here with #36. To date, this library only generated types and this would require generating a runtime.

 I’ll think on this.

**Comment by sachinraja** [comment]
This API seems kind of similar to https://github.com/aspida/pathpida. I really like how nextjs-routes is just a drop-in solution right now (with no runtime!).

**Comment by IRediTOTO** [comment]
> This API seems kind of similar to https://github.com/aspida/pathpida. I really like how nextjs-routes is just a drop-in solution right now (with no runtime impact!).

Look like this is what i am searching.

**Comment by tatethurston** [comment]
Yeah pathpida is great if you want a runtime

**Comment by tatethurston** [comment]
If there is sufficient interest in a runtime I’ll look into it for nextjs-routes but pathpida seems to fill this need.

**Comment by tatethurston** [comment]
This is released in [v0.0.21](https://github.com/tatethurston/nextjs-routes/releases/tag/v0.0.21)
